### PR TITLE
fix: use HttpLocaleReolver to resolve locale

### LINKED ISSUE #288: For ThymeLeafViewsRenderer use HttpLocaleResolver instead of request directly for resolving the locale
### Feature description

Currently the [ThymeLeafViewsRenderer](https://github.com/micronaut-projects/micronaut-views/blob/master/views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java#L79) uses `request.locale` to apply the locale to the context. I believe to remain consistent, it should instead use `HttpLocaleResolver.resolve(request)`. This is necessary to allow users to change their locale via a cookie or some other means and have the change take effect without forcing the user to change their browser's language settings.

### PR DESCRIPTION
**Pull request by karowan**

### CODE REVIEW

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java [hunk 1]
```diff
@@ -52,20 +53,24 @@
...
     protected ResourceLoader resourceLoader;
 
     /**
      * @param templateResolver   The template resolver
      * @param templateEngine     The template engine
      * @param resourceLoader     The resource loader
+     * @param httpLocaleResolver The locale resolver
      */
     @Inject
     public ThymeleafViewsRenderer(AbstractConfigurableTemplateResolver templateResolver,
```
Comment (sdelamo) [inline views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java:66]: This is a breaking change you need to leave the old constructor and flag it as `@Deprecated` and create a new constructor and annotate the new one with `@Inject`. Leave the old one without `@Inject`annotation.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java (hunk 1 above)
Comment (karowan) [reply views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java:66]: Fixed in the most recent push. In order to remain consistent with the previous logic, the deprecated constructor passes in an HttpLocaleResolver which only looks at the request's locale and defaults to US.

Comment (sdelamo) [reply views-thymeleaf/src/main/java/io/micronaut/views/thymeleaf/ThymeleafViewsRenderer.java:67]: thanks for doing the change.

**Comment by sdelamo** [comment]
Thanks for the contribution. I have added a javadoc `@deprecated` entry and merged your PR here: https://github.com/micronaut-projects/micronaut-views/pull/295

**Pull request closed without merging**